contracts/stellar-multicall/src/contract.rs

```diff
@@ -0,0 +1,33 @@
+use soroban_sdk::{contract, contractimpl, Env, Val, Vec};
+
+use crate::error::ContractError;
+use crate::interface::MulticallInterface;
+use crate::types::FunctionCall;
+
+#[contract]
+pub struct Multicall;
+
+#[contractimpl]
+impl Multicall {
+    pub const fn __constructor(_env: &Env) {}
+}
+
+#[contractimpl]
+impl MulticallInterface for Multicall {
+    fn multicall(env: &Env, function_calls: Vec<FunctionCall>) -> Result<Vec<Val>, ContractError> {
+        let mut results = Vec::new(env);
+
+        for FunctionCall {
+            contract,
+            function,
+            args,
+        } in function_calls.into_iter()
+        {
+            let result: Val = env.invoke_contract(&contract, &function, args.clone());
```

Comment:
multicall should support invoking methods that call `require_auth` on some `approver`. Explicitly add `approver.require_auth()` here so sub-invocation authorization is nested